pkg/controllers/metrics/node/controller.go

```diff
@@ -128,15 +128,15 @@ func labelNames() []string {
 }
 
 type Controller struct {
-	KubeClient      client.Client
-	LabelCollection map[types.NamespacedName][]prometheus.Labels
+	kubeClient      client.Client
+	labelCollection *labelsMap
```

Comment:
Is it possible to just use a sync.map here?

Reply:
Sounds good to me, but I would change labelsMap to use a sync.Map internally to preserve the type safety in the controller.

Reply:
I think you could probably get away without the type safety. Usage of these maps are quite limited in their scope -- especially if we make them private like you've done here.
 
- pod.LabelsMap is used twice, once to get and another to set a value
- node.LabelCollection is used three times, once to get and twice to set a value

It would be a different story if these maps were exposed or used broadly. You might notice across your PRs so far that I'm really focused on minimizing complexity and the number of concepts in the codebase. Each new abstraction introduces new concepts that must be understood and used correctly by a huge community of engineers who will read/write for the future of the codebase.